### PR TITLE
feat(TCOMP-1892): Upgrade Spark to version 3.0.1

### PR DESCRIPTION
**Pull request by iemejia**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
To align versions with Pipeline Designer

### What does this PR adds (design/code thoughts)?
Upgrades Spark version to version 3 and Scala to version 2.12

R: @undx

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -218,8 +218,8 @@
     <!-- should follow beam-sdk-java-core version -->
     <joda.time.version>2.10.5</joda.time.version>
 
-    <spark.version>2.4.7</spark.version>
-    <spark-scala.version>2.11</spark-scala.version>
+    <spark.version>3.0.1</spark.version>
```
Comment (iemejia) [inline pom.xml:221]: Notice that we could be using a more recent version of Spark here but it is probably a good idea to be aligned with the Spark version that the Data Processing Platform team targets.

**Review by undx — APPROVED** [review]
LGTM
still some pending tests on connectors repos but more relevant from component-runtime bump.
Already merged on my side.

**Comment by undx** [comment]
Already merged beside, closing.

**Comment by iemejia** [comment]
Thanks @undx ! Hope this make it into the next TCK release.

**Pull request closed without merging**